src/RestSharp/Request/RequestContent.cs

```diff
@@ -36,38 +36,66 @@ class RequestContent : IDisposable {
         _parameters = new RequestParameters(_request.Parameters.Union(_client.DefaultParameters));
     }
 
-    public HttpContent BuildContent() {
-        AddFiles();
-        var postParameters = _parameters.GetContentParameters(_request.Method).ToArray();
-        AddBody(postParameters.Length > 0);
-        AddPostParameters(postParameters);
+    public HttpContent BuildContent() 
+    {
+     if (_request.AlwaysMultipartFormData && _request.AlwaysSingleFileAsContent) 
```

Comment:
Maybe it would be good to fail fast by moving the mutual exclusivity check to the property setters?

Reply:
Calling validation in different setters will complicate the readability of the code. And the validation of adding parameters is difficult to implement because the user can use ``` request.Parameters.AddParameter ```